torch/csrc/jit/script/module.h

```diff
@@ -481,6 +481,25 @@ struct Module {
     }
     fn(*this);
   }
+  void train(bool on = true) {
+    for (auto& submod : get_modules()) {
+      submod.value().module->train(on);
+    }
+    auto t = autograd::make_variable(at::full({}, on ? 1 : 0, at::kLong));
```

Comment:
You can replace
```
auto t = autograd::make_variable(at::full({}, on ? 1 : 0, at::kLong));
```
with
```
auto t = torch::tensor(on ? 1 : 0, at::kLong);
```

To clarify: `torch::` factory functions create variables while `at::` functions create tensors. `torch::tensor`/`at::tensor` is like `torch.tensor` in python (creates a tensor with the values you give it)`

Reply:
And then just embed it in the `register_parameter` call:
```
register_parameter("training", torch::tensor(on ? 1 : 0, at::kLong), /*is_buffer=*/true);
```